flytekit/image_spec/image_spec.py

```diff
@@ -111,15 +123,15 @@ def build(cls, image_spec: ImageSpec):
             click.secho(f"Image {image_spec.image_name()} found. Skip building.", fg="blue")
 
 
-@lru_cache(maxsize=None)
+@lru_cache
 def calculate_hash_from_image_spec(image_spec: ImageSpec):
     """
     Calculate the hash from the image spec.
     """
+    spec = copy(image_spec)
```

Comment:
can you add a comment saying why we need to copy this?

Reply:
done